java/de/yessoft/android/mohammad/application/app.java

```diff
@@ -0,0 +1,58 @@
+package de.yessoft.android.mohammad.application;
+
+import android.app.Application;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.os.Build;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.FirebaseFirestoreSettings;
+
+public class app extends Application {
+
+    public static final String CHANNEL_1_ID = "channel1";
+    public static final String CHANNEL_2_ID = "channel2";
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        // region Enable Offline Data
+        FirebaseFirestoreSettings settings = new FirebaseFirestoreSettings
+                .Builder()
+                .setPersistenceEnabled(true)
+                .build();
+
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+        db.setFirestoreSettings(settings);
+        // endregion
+
+        // region Notification
+        createNotificationChannels();
+        // endregion
+    }
+
+    private void createNotificationChannels()
```

Comment:
same as firebase , we should have notification service interface ,
and implementation for this interface
in the constructor pass the interface only
our class should not know about the implementation

Reply:
Replied Below

Reply:
DI of cores !